hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AInputPolicies.java

```diff
@@ -79,13 +76,18 @@ public static Collection<Object[]> data() {
     });
   }
 
-  @Test
-  public void testInputPolicies() throws Throwable {
-    Assert.assertEquals(
-        String.format("calculateRequestLimit(%s, %d, %d, %d, %d)",
-            policy, targetPos, length, contentLength, readahead),
+  @MethodSource("data")
+  @ParameterizedTest
+  public void testInputPolicies(S3AInputPolicy pPolicy,
+      long pTargetPos, long pLength, long pContentLength,
+      long pReadahead, long pExpectedLimit) throws Throwable {
+    initTestS3AInputPolicies(pPolicy, pTargetPos, pLength, pContentLength,
```

Comment:
@steveloughran Regarding the Parameterized test issue, we have found some solutions, although it will significantly increase the workload. We can call initTestS3AInputPolicies in each unit test and explicitly call the setup method in the init method, which ensures that the tests pass.
cc: @cnauroth